### DISCUSSION TITLE
Nextjs Image component stops working when configuring basePath

### DISCUSSION THREAD

**Discussion by omerman** [vercel/next.js#34173] [Help] [answered]
### Run `next info` (available from version 12.0.8 and up)

      Operating System:
      Platform: darwin
      Arch: x64
    Binaries:
      Node: 17.3.0
      npm: 8.3.0
      Yarn: 1.22.4
      pnpm: N/A
    Relevant packages:
      next: 12.0.10
      react: 17.0.2
      react-dom: 17.0.2

### What version of Next.js are you using?

12.0.10

### What version of Node.js are you using?

17.3.0

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

next start / next dev

### Describe the Bug

Adding basePath to nextjs config, seems to make the public folder assets unavailable.
The basePath prefix is added, but the images are not shown.

The attached github repo is a created nextjs from docs `yarn create next-app --typescript` + added a one liner of basePath
https://github.com/omerman/nextjs-image-basePath-bug
**As you can see in this repo, the vercel icon at the bottom of the page does not show**

### Expected Behavior

The basePath should work well with Image component

### To Reproduce

As mentioned, 
1. `yarn create next-app --typescript`
2. added a one liner of basePath
3. run the app in either dev or prod mode (build & start)

See https://github.com/omerman/nextjs-image-basePath-bug
**As you can see in this repo, the vercel icon at the bottom of the page does not show**

**Comment by omerman** [comment]
![image](https://user-images.githubusercontent.com/6780529/153356683-53a9e16b-81d7-4696-b10f-03690c0d29eb.png)

**Comment by omerman** [comment]
![image](https://user-images.githubusercontent.com/6780529/153357356-ce7bc6d6-033d-419f-b8b5-2ebee1601630.png)

**Comment by balazsorban44** [comment, marked as answer]
Hi, when using `next/image`, you will have to apply the `basePath` as documented here:

https://nextjs.org/docs/api-reference/next.config.js/basepath#images

**Comment by omerman** [comment]
@balazsorban44 
First of all thank you for the quick response. adding the prefix myself makes It work.

But I'll tell you why it is counter intuitive.. because nextjs seems to add the prefix by itself to the Image component,
meaning if you run the example you will see that it adds the prefix, but even-though the prefix is added automatically it fails to display it.

nextjs should ideally add the prefix automatically for Image components and it should work out of the box, or at least not add it and it will be easier for me to understand that I should.. Dont you agree?

**Reply by balazsorban44** [reply]
I think the reason is that images could come from remote sources as well through a custom loader or when a `path` is given in `images` in `next.config.js` and there is no guarantee that the same `basePath` is necessary.

**Reply by omerman** [reply]
I understand what you are saying, but I think that it is a pretty good indicator that if the BasePath is configured as /path and under my public folder there is a folder called path, I want the asset under the nextjs's /path. What do you think?

**Comment by raychunghk** [comment]
If you setup your basePath in your next.config.js, you should avoid hard-coding the basepath in your Image component.
You can use the basePath config variable as below.

next.config.js

      `const nextConfig = {
      basePath: '/absproxy/3000',
      assetPrefix: '/absproxy/3000',
      reactStrictMode: true,
      swcMinify: true,
      trailingSlash: true,
      
      }
      
      module.exports = nextConfig`
 



Then use the basePath as below:
        `import Head from 'next/head';
         
        import Image from 'next/image';
        import flower from '../public/images/flower.jpg'
        import { basePath } from '../next.config';
        
        export default function Home() {
          return (
            <div className={styles.container}>
             
        
              <main>
                <Image
                  src={`${basePath}/images/flower.jpg`} 
                  height={144}  
                  width={144} 
                  alt="Your Name"
                />
         
              </main>
             
         
            </div>
          )
        }`

I also created the answer in stackoverflow in regard the same matter.
[answer](https://stackoverflow.com/a/76197261/1914557)
